// cashfree-pg-sdk-nodejs/dist/src/interface/CFResponses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFPaymentLinkOrdersRespnse = exports.CFCancelledLinkResponse = exports.CFLinkResponse = exports.CFCryptogramResponse = exports.CFInstrumentsResponse = exports.CFAllInstrumentsResponse = exports.CFSettlementsResponse = exports.CFAllRefundsResponse = exports.CFRefundResponse = exports.CFPaymentEntityResponse = exports.CFPaymentsForOrderResponse = exports.CFPayResponse = exports.CFOrderResponse = void 0;
class CFOrderResponse {
    constructor(cfOrder, cfHeaders = {}) {
        this.cfOrder = cfOrder;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFOrderResponse = CFOrderResponse;
class CFPayResponse {
    constructor(cfOrderPayResponse, cfHeaders = {}) {
        this.cfOrderPayResponse = cfOrderPayResponse;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFPayResponse = CFPayResponse;
class CFPaymentsForOrderResponse {
    constructor(cfPaymentsEntities, cfHeaders = {}) {
        this.cfPaymentsEntities = cfPaymentsEntities;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFPaymentsForOrderResponse = CFPaymentsForOrderResponse;
class CFPaymentEntityResponse {
    constructor(cfPaymentsEntity, cfHeaders = {}) {
        this.cfPaymentsEntity = cfPaymentsEntity;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFPaymentEntityResponse = CFPaymentEntityResponse;
class CFRefundResponse {
    constructor(cfRefund, cfHeaders = {}) {
        this.cfRefund = cfRefund;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFRefundResponse = CFRefundResponse;
class CFAllRefundsResponse {
    constructor(cfRefunds, cfHeaders = {}) {
        this.cfRefunds = cfRefunds;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFAllRefundsResponse = CFAllRefundsResponse;
class CFSettlementsResponse {
    constructor(cfSettlementsEntity, cfHeaders = {}) {
        this.cfSettlementsEntity = cfSettlementsEntity;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFSettlementsResponse = CFSettlementsResponse;
class CFAllInstrumentsResponse {
    constructor(cfInstruments, cfHeaders = {}) {
        this.cfInstruments = cfInstruments;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFAllInstrumentsResponse = CFAllInstrumentsResponse;
class CFInstrumentsResponse {
    constructor(cfInstrument, cfHeaders = {}) {
        this.cfInstrument = cfInstrument;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFInstrumentsResponse = CFInstrumentsResponse;
class CFCryptogramResponse {
    constructor(cfCryptogram, cfHeaders = {}) {
        this.cfCryptogram = cfCryptogram;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFCryptogramResponse = CFCryptogramResponse;
class CFLinkResponse {
    constructor(cfLink, cfHeaders = {}) {
        this.cfLink = cfLink;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFLinkResponse = CFLinkResponse;
class CFCancelledLinkResponse {
    constructor(cfLink, cfHeaders = {}) {
        this.cfLink = cfLink;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFCancelledLinkResponse = CFCancelledLinkResponse;
class CFPaymentLinkOrdersRespnse {
    constructor(cfLinkOrders, cfHeaders = {}) {
        this.cfLinkOrders = cfLinkOrders;
        this.cfHeaders = cfHeaders;
    }
}
exports.CFPaymentLinkOrdersRespnse = CFPaymentLinkOrdersRespnse;
