// cashfree-pg-sdk-nodejs/dist/src/model/cFAuthorizationInPaymentsEntity.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFAuthorizationInPaymentsEntity = void 0;
/**
* The authorization details are present for payments which go through the preauthorization workflow. Or else this parameter will be null.
*/
class CFAuthorizationInPaymentsEntity {
    static getAttributeTypeMap() {
        return CFAuthorizationInPaymentsEntity.attributeTypeMap;
    }
}
exports.CFAuthorizationInPaymentsEntity = CFAuthorizationInPaymentsEntity;
CFAuthorizationInPaymentsEntity.discriminator = undefined;
CFAuthorizationInPaymentsEntity.attributeTypeMap = [
    {
        "name": "action",
        "baseName": "action",
        "type": "CFAuthorizationInPaymentsEntity.ActionEnum"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "CFAuthorizationInPaymentsEntity.StatusEnum"
    },
    {
        "name": "capturedAmount",
        "baseName": "captured_amount",
        "type": "number"
    },
    {
        "name": "startTime",
        "baseName": "start_time",
        "type": "string"
    },
    {
        "name": "endTime",
        "baseName": "end_time",
        "type": "string"
    },
    {
        "name": "approveBy",
        "baseName": "approve_by",
        "type": "string"
    }
];
(function (CFAuthorizationInPaymentsEntity) {
    let ActionEnum;
    (function (ActionEnum) {
        ActionEnum[ActionEnum["Capture"] = 'CAPTURE'] = "Capture";
        ActionEnum[ActionEnum["Void"] = 'VOID'] = "Void";
    })(ActionEnum = CFAuthorizationInPaymentsEntity.ActionEnum || (CFAuthorizationInPaymentsEntity.ActionEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Success"] = 'SUCCESS'] = "Success";
        StatusEnum[StatusEnum["Pending"] = 'PENDING'] = "Pending";
    })(StatusEnum = CFAuthorizationInPaymentsEntity.StatusEnum || (CFAuthorizationInPaymentsEntity.StatusEnum = {}));
})(CFAuthorizationInPaymentsEntity = exports.CFAuthorizationInPaymentsEntity || (exports.CFAuthorizationInPaymentsEntity = {}));
