// cashfree-pg-sdk-nodejs/dist/src/model/cFAuthorizationRequest.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFAuthorizationRequest = void 0;
class CFAuthorizationRequest {
    static getAttributeTypeMap() {
        return CFAuthorizationRequest.attributeTypeMap;
    }
}
exports.CFAuthorizationRequest = CFAuthorizationRequest;
CFAuthorizationRequest.discriminator = undefined;
CFAuthorizationRequest.attributeTypeMap = [
    {
        "name": "action",
        "baseName": "action",
        "type": "CFAuthorizationRequest.ActionEnum"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "number"
    }
];
(function (CFAuthorizationRequest) {
    let ActionEnum;
    (function (ActionEnum) {
        ActionEnum[ActionEnum["Capture"] = 'CAPTURE'] = "Capture";
        ActionEnum[ActionEnum["Void"] = 'VOID'] = "Void";
    })(ActionEnum = CFAuthorizationRequest.ActionEnum || (CFAuthorizationRequest.ActionEnum = {}));
})(CFAuthorizationRequest = exports.CFAuthorizationRequest || (exports.CFAuthorizationRequest = {}));
