// cashfree-pg-sdk-nodejs/dist/src/model/cFCardEMI.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFCardEMI = void 0;
class CFCardEMI {
    constructor() {
        /**
        * The channel for card payments will always be \"link\"
        */
        this['channel'] = "link";
    }
    static getAttributeTypeMap() {
        return CFCardEMI.attributeTypeMap;
    }
}
exports.CFCardEMI = CFCardEMI;
CFCardEMI.discriminator = undefined;
CFCardEMI.attributeTypeMap = [
    {
        "name": "channel",
        "baseName": "channel",
        "type": "CFCardEMI.ChannelEnum"
    },
    {
        "name": "cardNumber",
        "baseName": "card_number",
        "type": "string"
    },
    {
        "name": "cardHolderName",
        "baseName": "card_holder_name",
        "type": "string"
    },
    {
        "name": "cardExpiryMm",
        "baseName": "card_expiry_mm",
        "type": "string"
    },
    {
        "name": "cardExpiryYy",
        "baseName": "card_expiry_yy",
        "type": "string"
    },
    {
        "name": "cardCvv",
        "baseName": "card_cvv",
        "type": "string"
    },
    {
        "name": "cardAlias",
        "baseName": "card_alias",
        "type": "string"
    },
    {
        "name": "cardBankName",
        "baseName": "card_bank_name",
        "type": "CFCardEMI.CardBankNameEnum"
    },
    {
        "name": "emiTenure",
        "baseName": "emi_tenure",
        "type": "number"
    }
];
(function (CFCardEMI) {
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum[ChannelEnum["Link"] = 'link'] = "Link";
    })(ChannelEnum = CFCardEMI.ChannelEnum || (CFCardEMI.ChannelEnum = {}));
    let CardBankNameEnum;
    (function (CardBankNameEnum) {
        CardBankNameEnum[CardBankNameEnum["Kotak"] = 'Kotak'] = "Kotak";
        CardBankNameEnum[CardBankNameEnum["Icici"] = 'ICICI'] = "Icici";
        CardBankNameEnum[CardBankNameEnum["Rbl"] = 'RBL'] = "Rbl";
        CardBankNameEnum[CardBankNameEnum["Bob"] = 'BOB'] = "Bob";
        CardBankNameEnum[CardBankNameEnum["StandardChartered"] = 'Standard Chartered'] = "StandardChartered";
        CardBankNameEnum[CardBankNameEnum["Hdfc"] = 'HDFC'] = "Hdfc";
    })(CardBankNameEnum = CFCardEMI.CardBankNameEnum || (CFCardEMI.CardBankNameEnum = {}));
})(CFCardEMI = exports.CFCardEMI || (exports.CFCardEMI = {}));
