// cashfree-pg-sdk-nodejs/dist/src/model/cFCard.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFCard = void 0;
class CFCard {
    static getAttributeTypeMap() {
        return CFCard.attributeTypeMap;
    }
}
exports.CFCard = CFCard;
CFCard.discriminator = undefined;
CFCard.attributeTypeMap = [
    {
        "name": "channel",
        "baseName": "channel",
        "type": "string"
    },
    {
        "name": "cardNumber",
        "baseName": "card_number",
        "type": "string"
    },
    {
        "name": "cardHolderName",
        "baseName": "card_holder_name",
        "type": "string"
    },
    {
        "name": "cardExpiryMm",
        "baseName": "card_expiry_mm",
        "type": "string"
    },
    {
        "name": "cardExpiryYy",
        "baseName": "card_expiry_yy",
        "type": "string"
    },
    {
        "name": "cardCvv",
        "baseName": "card_cvv",
        "type": "string"
    },
    {
        "name": "instrumentId",
        "baseName": "instrument_id",
        "type": "string"
    },
    {
        "name": "cryptogram",
        "baseName": "cryptogram",
        "type": "string"
    },
    {
        "name": "tokenRequestorId",
        "baseName": "token_requestor_id",
        "type": "string"
    },
    {
        "name": "cardDisplay",
        "baseName": "card_display",
        "type": "string"
    },
    {
        "name": "cardAlias",
        "baseName": "card_alias",
        "type": "string"
    },
    {
        "name": "cardBankName",
        "baseName": "card_bank_name",
        "type": "CFCard.CardBankNameEnum"
    },
    {
        "name": "emiTenure",
        "baseName": "emi_tenure",
        "type": "number"
    }
];
(function (CFCard) {
    let CardBankNameEnum;
    (function (CardBankNameEnum) {
        CardBankNameEnum[CardBankNameEnum["Kotak"] = 'Kotak'] = "Kotak";
        CardBankNameEnum[CardBankNameEnum["Icici"] = 'ICICI'] = "Icici";
        CardBankNameEnum[CardBankNameEnum["Rbl"] = 'RBL'] = "Rbl";
        CardBankNameEnum[CardBankNameEnum["Bob"] = 'BOB'] = "Bob";
        CardBankNameEnum[CardBankNameEnum["StandardChartered"] = 'Standard Chartered'] = "StandardChartered";
        CardBankNameEnum[CardBankNameEnum["Test"] = 'Test'] = "Test";
    })(CardBankNameEnum = CFCard.CardBankNameEnum || (CFCard.CardBankNameEnum = {}));
})(CFCard = exports.CFCard || (exports.CFCard = {}));
