// cashfree-pg-sdk-nodejs/dist/src/model/cFCardlessEMI.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFCardlessEMI = void 0;
class CFCardlessEMI {
    static getAttributeTypeMap() {
        return CFCardlessEMI.attributeTypeMap;
    }
}
exports.CFCardlessEMI = CFCardlessEMI;
CFCardlessEMI.discriminator = undefined;
CFCardlessEMI.attributeTypeMap = [
    {
        "name": "channel",
        "baseName": "channel",
        "type": "CFCardlessEMI.ChannelEnum"
    },
    {
        "name": "provider",
        "baseName": "provider",
        "type": "CFCardlessEMI.ProviderEnum"
    },
    {
        "name": "phone",
        "baseName": "phone",
        "type": "string"
    }
];
(function (CFCardlessEMI) {
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum[ChannelEnum["Link"] = 'link'] = "Link";
    })(ChannelEnum = CFCardlessEMI.ChannelEnum || (CFCardlessEMI.ChannelEnum = {}));
    let ProviderEnum;
    (function (ProviderEnum) {
        ProviderEnum[ProviderEnum["Flexmoney"] = 'flexmoney'] = "Flexmoney";
        ProviderEnum[ProviderEnum["Zestmoney"] = 'zestmoney'] = "Zestmoney";
    })(ProviderEnum = CFCardlessEMI.ProviderEnum || (CFCardlessEMI.ProviderEnum = {}));
})(CFCardlessEMI = exports.CFCardlessEMI || (exports.CFCardlessEMI = {}));
