// cashfree-pg-sdk-nodejs/dist/src/model/cFError.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFError = void 0;
class CFError {
    static getAttributeTypeMap() {
        return CFError.attributeTypeMap;
    }
}
exports.CFError = CFError;
CFError.discriminator = undefined;
CFError.attributeTypeMap = [
    {
        "name": "message",
        "baseName": "message",
        "type": "string"
    },
    {
        "name": "code",
        "baseName": "code",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CFError.TypeEnum"
    }
];
(function (CFError) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["InvalidRequestError"] = 'invalid_request_error'] = "InvalidRequestError";
        TypeEnum[TypeEnum["AuthenticationError"] = 'authentication_error'] = "AuthenticationError";
        TypeEnum[TypeEnum["RateLimitError"] = 'rate_limit_error'] = "RateLimitError";
        TypeEnum[TypeEnum["ValidationError"] = 'validation_error'] = "ValidationError";
        TypeEnum[TypeEnum["ApiError"] = 'api_error'] = "ApiError";
    })(TypeEnum = CFError.TypeEnum || (CFError.TypeEnum = {}));
})(CFError = exports.CFError || (exports.CFError = {}));
