// cashfree-pg-sdk-nodejs/dist/src/model/cFOrderPayResponse.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFOrderPayResponse = void 0;
class CFOrderPayResponse {
    static getAttributeTypeMap() {
        return CFOrderPayResponse.attributeTypeMap;
    }
}
exports.CFOrderPayResponse = CFOrderPayResponse;
CFOrderPayResponse.discriminator = undefined;
CFOrderPayResponse.attributeTypeMap = [
    {
        "name": "cfPaymentId",
        "baseName": "cf_payment_id",
        "type": "number"
    },
    {
        "name": "paymentMethod",
        "baseName": "payment_method",
        "type": "CFOrderPayResponse.PaymentMethodEnum"
    },
    {
        "name": "channel",
        "baseName": "channel",
        "type": "CFOrderPayResponse.ChannelEnum"
    },
    {
        "name": "action",
        "baseName": "action",
        "type": "CFOrderPayResponse.ActionEnum"
    },
    {
        "name": "data",
        "baseName": "data",
        "type": "CFOrderPayData"
    }
];
(function (CFOrderPayResponse) {
    let PaymentMethodEnum;
    (function (PaymentMethodEnum) {
        PaymentMethodEnum[PaymentMethodEnum["Netbanking"] = 'netbanking'] = "Netbanking";
        PaymentMethodEnum[PaymentMethodEnum["Card"] = 'card'] = "Card";
        PaymentMethodEnum[PaymentMethodEnum["Upi"] = 'upi'] = "Upi";
        PaymentMethodEnum[PaymentMethodEnum["App"] = 'app'] = "App";
        PaymentMethodEnum[PaymentMethodEnum["CardlessEmi"] = 'cardless_emi'] = "CardlessEmi";
        PaymentMethodEnum[PaymentMethodEnum["Paylater"] = 'paylater'] = "Paylater";
    })(PaymentMethodEnum = CFOrderPayResponse.PaymentMethodEnum || (CFOrderPayResponse.PaymentMethodEnum = {}));
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum[ChannelEnum["Link"] = 'link'] = "Link";
        ChannelEnum[ChannelEnum["Collect"] = 'collect'] = "Collect";
        ChannelEnum[ChannelEnum["Qrcode"] = 'qrcode'] = "Qrcode";
    })(ChannelEnum = CFOrderPayResponse.ChannelEnum || (CFOrderPayResponse.ChannelEnum = {}));
    let ActionEnum;
    (function (ActionEnum) {
        ActionEnum[ActionEnum["Link"] = 'link'] = "Link";
        ActionEnum[ActionEnum["Custom"] = 'custom'] = "Custom";
        ActionEnum[ActionEnum["Form"] = 'form'] = "Form";
    })(ActionEnum = CFOrderPayResponse.ActionEnum || (CFOrderPayResponse.ActionEnum = {}));
})(CFOrderPayResponse = exports.CFOrderPayResponse || (exports.CFOrderPayResponse = {}));
