// cashfree-pg-sdk-nodejs/dist/src/model/cFPaylater.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFPaylater = void 0;
class CFPaylater {
    static getAttributeTypeMap() {
        return CFPaylater.attributeTypeMap;
    }
}
exports.CFPaylater = CFPaylater;
CFPaylater.discriminator = undefined;
CFPaylater.attributeTypeMap = [
    {
        "name": "channel",
        "baseName": "channel",
        "type": "string"
    },
    {
        "name": "provider",
        "baseName": "provider",
        "type": "CFPaylater.ProviderEnum"
    },
    {
        "name": "phone",
        "baseName": "phone",
        "type": "string"
    }
];
(function (CFPaylater) {
    let ProviderEnum;
    (function (ProviderEnum) {
        ProviderEnum[ProviderEnum["Kotak"] = 'kotak'] = "Kotak";
        ProviderEnum[ProviderEnum["Flexipay"] = 'flexipay'] = "Flexipay";
        ProviderEnum[ProviderEnum["Zestmoney"] = 'zestmoney'] = "Zestmoney";
        ProviderEnum[ProviderEnum["Lazypay"] = 'lazypay'] = "Lazypay";
        ProviderEnum[ProviderEnum["Olapostpaid"] = 'olapostpaid'] = "Olapostpaid";
    })(ProviderEnum = CFPaylater.ProviderEnum || (CFPaylater.ProviderEnum = {}));
})(CFPaylater = exports.CFPaylater || (exports.CFPaylater = {}));
