// cashfree-pg-sdk-nodejs/dist/src/model/cFUPI.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFUPI = void 0;
class CFUPI {
    static getAttributeTypeMap() {
        return CFUPI.attributeTypeMap;
    }
}
exports.CFUPI = CFUPI;
CFUPI.discriminator = undefined;
CFUPI.attributeTypeMap = [
    {
        "name": "channel",
        "baseName": "channel",
        "type": "CFUPI.ChannelEnum"
    },
    {
        "name": "upiId",
        "baseName": "upi_id",
        "type": "string"
    },
    {
        "name": "authorizeOnly",
        "baseName": "authorize_only",
        "type": "boolean"
    },
    {
        "name": "authorization",
        "baseName": "authorization",
        "type": "CFUPIAuthorizeDetails"
    }
];
(function (CFUPI) {
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum[ChannelEnum["Link"] = 'link'] = "Link";
        ChannelEnum[ChannelEnum["Collect"] = 'collect'] = "Collect";
        ChannelEnum[ChannelEnum["Qrcode"] = 'qrcode'] = "Qrcode";
    })(ChannelEnum = CFUPI.ChannelEnum || (CFUPI.ChannelEnum = {}));
})(CFUPI = exports.CFUPI || (exports.CFUPI = {}));
