// cashfree-pg-sdk-nodejs/dist/src/api/apis.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIS = exports.HttpError = void 0;
__exportStar(require("./ordersApi"), exports);
const ordersApi_1 = require("./ordersApi");
__exportStar(require("./paymentLinksApi"), exports);
const paymentLinksApi_1 = require("./paymentLinksApi");
__exportStar(require("./paymentsApi"), exports);
const paymentsApi_1 = require("./paymentsApi");
__exportStar(require("./refundsApi"), exports);
const refundsApi_1 = require("./refundsApi");
__exportStar(require("./settlementsApi"), exports);
const settlementsApi_1 = require("./settlementsApi");
__exportStar(require("./tokenVaultApi"), exports);
const tokenVaultApi_1 = require("./tokenVaultApi");
class HttpError extends Error {
    constructor(response, body, statusCode) {
        super('HTTP request failed');
        this.response = response;
        this.body = body;
        this.statusCode = statusCode;
        this.name = 'HttpError';
    }
}
exports.HttpError = HttpError;
exports.APIS = [ordersApi_1.OrdersApi, paymentLinksApi_1.PaymentLinksApi, paymentsApi_1.PaymentsApi, refundsApi_1.RefundsApi, settlementsApi_1.SettlementsApi, tokenVaultApi_1.TokenVaultApi];
