// cashfree-pg-sdk-nodejs/dist/src/api/paymentLinksApi.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentLinksApi = exports.PaymentLinksApiApiKeys = void 0;
const request_1 = __importDefault(require("request"));
const models_1 = require("../model/models");
const apis_1 = require("./apis");
let defaultBasePath = 'https://sandbox.cashfree.com/pg';
// ===============================================
// This file is autogenerated - Please do not edit
// ===============================================
var PaymentLinksApiApiKeys;
(function (PaymentLinksApiApiKeys) {
})(PaymentLinksApiApiKeys = exports.PaymentLinksApiApiKeys || (exports.PaymentLinksApiApiKeys = {}));
class PaymentLinksApi {
    constructor(basePathOrUsername, password, basePath) {
        this._basePath = defaultBasePath;
        this._defaultHeaders = {};
        this._useQuerystring = false;
        this.authentications = {
            'default': new models_1.VoidAuth(),
        };
        this.interceptors = [];
        if (password) {
            if (basePath) {
                this.basePath = basePath;
            }
        }
        else {
            if (basePathOrUsername) {
                this.basePath = basePathOrUsername;
            }
        }
    }
    set useQuerystring(value) {
        this._useQuerystring = value;
    }
    set basePath(basePath) {
        this._basePath = basePath;
    }
    set defaultHeaders(defaultHeaders) {
        this._defaultHeaders = defaultHeaders;
    }
    get defaultHeaders() {
        return this._defaultHeaders;
    }
    get basePath() {
        return this._basePath;
    }
    setDefaultAuthentication(auth) {
        this.authentications.default = auth;
    }
    setApiKey(key, value) {
        this.authentications[PaymentLinksApiApiKeys[key]].apiKey = value;
    }
    addInterceptor(interceptor) {
        this.interceptors.push(interceptor);
    }
    /**
     * Use this API to cancel a payment link. No further payments can be done against a cancelled link. Only a link in ACTIVE status can be cancelled.
     * @summary Cancel Payment Link
     * @param xClientId
     * @param xClientSecret
     * @param linkId
     * @param xApiVersion
     */
    cancelPaymentLink(xClientId, xClientSecret, linkId, xApiVersion, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/links/{link_id}/cancel'
                .replace('{' + 'link_id' + '}', encodeURIComponent(String(linkId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling cancelPaymentLink.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling cancelPaymentLink.');
            }
            // verify required parameter 'linkId' is not null or undefined
            if (linkId === null || linkId === undefined) {
                throw new Error('Required parameter linkId was null or undefined when calling cancelPaymentLink.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'POST', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "CFLinkCancelledResponse");
                                resolve({ cfHeaders: response.headers, cfLink: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    /**
     * Use this API to create a new payment link. The created payment link url will be available in the API response parameter link_url.
     * @summary Create Payment Link
     * @param xClientId
     * @param xClientSecret
     * @param xApiVersion
     * @param xIdempotencyReplayed
     * @param xIdempotencyKey
     * @param xRequestId
     * @param cFLinkRequest
     */
    createPaymentLink(xClientId, xClientSecret, xApiVersion, xIdempotencyReplayed, xIdempotencyKey, xRequestId, cFLinkRequest, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/links';
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling createPaymentLink.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling createPaymentLink.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            localVarHeaderParams['x-idempotency-replayed'] = models_1.ObjectSerializer.serialize(xIdempotencyReplayed, "boolean");
            localVarHeaderParams['x-idempotency-key'] = models_1.ObjectSerializer.serialize(xIdempotencyKey, "string");
            localVarHeaderParams['x-request-id'] = models_1.ObjectSerializer.serialize(xRequestId, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'POST', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, body: models_1.ObjectSerializer.serialize(cFLinkRequest, "CFLinkRequest"), timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "CFLink");
                                resolve({ cfHeaders: response.headers, cfLink: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    /**
     * Use this API to view all details and status of a payment link.
     * @summary Fetch Payment Link Details
     * @param xClientId
     * @param xClientSecret
     * @param linkId
     * @param xApiVersion
     * @param xIdempotencyReplayed
     * @param xIdempotencyKey
     * @param xRequestId
     */
    getPaymentLinkDetails(xClientId, xClientSecret, linkId, xApiVersion, xIdempotencyReplayed, xIdempotencyKey, xRequestId, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/links/{link_id}'
                .replace('{' + 'link_id' + '}', encodeURIComponent(String(linkId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling getPaymentLinkDetails.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling getPaymentLinkDetails.');
            }
            // verify required parameter 'linkId' is not null or undefined
            if (linkId === null || linkId === undefined) {
                throw new Error('Required parameter linkId was null or undefined when calling getPaymentLinkDetails.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            localVarHeaderParams['x-idempotency-replayed'] = models_1.ObjectSerializer.serialize(xIdempotencyReplayed, "boolean");
            localVarHeaderParams['x-idempotency-key'] = models_1.ObjectSerializer.serialize(xIdempotencyKey, "string");
            localVarHeaderParams['x-request-id'] = models_1.ObjectSerializer.serialize(xRequestId, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'GET', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "CFLink");
                                resolve({ cfHeaders: response.headers, cfLink: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    /**
     * Use this API to view all order details for a payment link.
     * @summary Get Orders for a Payment Link
     * @param xClientId
     * @param xClientSecret
     * @param linkId
     * @param xApiVersion
     * @param xIdempotencyReplayed
     * @param xIdempotencyKey
     * @param xRequestId
     */
    getPaymentLinkOrders(xClientId, xClientSecret, linkId, xApiVersion, xIdempotencyReplayed, xIdempotencyKey, xRequestId, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/links/{link_id}/orders'
                .replace('{' + 'link_id' + '}', encodeURIComponent(String(linkId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling getPaymentLinkOrders.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling getPaymentLinkOrders.');
            }
            // verify required parameter 'linkId' is not null or undefined
            if (linkId === null || linkId === undefined) {
                throw new Error('Required parameter linkId was null or undefined when calling getPaymentLinkOrders.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            localVarHeaderParams['x-idempotency-replayed'] = models_1.ObjectSerializer.serialize(xIdempotencyReplayed, "boolean");
            localVarHeaderParams['x-idempotency-key'] = models_1.ObjectSerializer.serialize(xIdempotencyKey, "string");
            localVarHeaderParams['x-request-id'] = models_1.ObjectSerializer.serialize(xRequestId, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'GET', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "Array<CFLinkOrders>");
                                resolve({ cfHeaders: response.headers, cfLinkOrders: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
}
exports.PaymentLinksApi = PaymentLinksApi;
