// cashfree-pg-sdk-nodejs/dist/src/api/refundsApi.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefundsApi = exports.RefundsApiApiKeys = void 0;
const request_1 = __importDefault(require("request"));
const models_1 = require("../model/models");
const apis_1 = require("./apis");
let defaultBasePath = 'https://sandbox.cashfree.com/pg';
// ===============================================
// This file is autogenerated - Please do not edit
// ===============================================
var RefundsApiApiKeys;
(function (RefundsApiApiKeys) {
})(RefundsApiApiKeys = exports.RefundsApiApiKeys || (exports.RefundsApiApiKeys = {}));
class RefundsApi {
    constructor(basePathOrUsername, password, basePath) {
        this._basePath = defaultBasePath;
        this._defaultHeaders = {};
        this._useQuerystring = false;
        this.authentications = {
            'default': new models_1.VoidAuth(),
        };
        this.interceptors = [];
        if (password) {
            if (basePath) {
                this.basePath = basePath;
            }
        }
        else {
            if (basePathOrUsername) {
                this.basePath = basePathOrUsername;
            }
        }
    }
    set useQuerystring(value) {
        this._useQuerystring = value;
    }
    set basePath(basePath) {
        this._basePath = basePath;
    }
    set defaultHeaders(defaultHeaders) {
        this._defaultHeaders = defaultHeaders;
    }
    get defaultHeaders() {
        return this._defaultHeaders;
    }
    get basePath() {
        return this._basePath;
    }
    setDefaultAuthentication(auth) {
        this.authentications.default = auth;
    }
    setApiKey(key, value) {
        this.authentications[RefundsApiApiKeys[key]].apiKey = value;
    }
    addInterceptor(interceptor) {
        this.interceptors.push(interceptor);
    }
    /**
     * Use this API to initiate refunds.
     * @summary Create Refund
     * @param xClientId
     * @param xClientSecret
     * @param orderId
     * @param xApiVersion
     * @param xIdempotencyReplayed
     * @param xIdempotencyKey
     * @param xRequestId
     * @param cFRefundRequest
     */
    createrefund(xClientId, xClientSecret, orderId, xApiVersion, xIdempotencyReplayed, xIdempotencyKey, xRequestId, cFRefundRequest, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/orders/{order_id}/refunds'
                .replace('{' + 'order_id' + '}', encodeURIComponent(String(orderId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling createrefund.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling createrefund.');
            }
            // verify required parameter 'orderId' is not null or undefined
            if (orderId === null || orderId === undefined) {
                throw new Error('Required parameter orderId was null or undefined when calling createrefund.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            localVarHeaderParams['x-idempotency-replayed'] = models_1.ObjectSerializer.serialize(xIdempotencyReplayed, "boolean");
            localVarHeaderParams['x-idempotency-key'] = models_1.ObjectSerializer.serialize(xIdempotencyKey, "string");
            localVarHeaderParams['x-request-id'] = models_1.ObjectSerializer.serialize(xRequestId, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'POST', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, body: models_1.ObjectSerializer.serialize(cFRefundRequest, "CFRefundRequest"), timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "CFRefund");
                                resolve({ cfHeaders: response.headers, cfRefund: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    /**
     * Use this API to fetch a specific refund processed on your Cashfree Account.
     * @summary Get Refund
     * @param xClientId
     * @param xClientSecret
     * @param orderId
     * @param refundId Refund Id of the refund you want to fetch.
     * @param xApiVersion
     * @param xIdempotencyReplayed
     * @param xIdempotencyKey
     * @param xRequestId
     */
    getRefund(xClientId, xClientSecret, orderId, refundId, xApiVersion, xIdempotencyReplayed, xIdempotencyKey, xRequestId, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/orders/{order_id}/refunds/{refund_id}'
                .replace('{' + 'order_id' + '}', encodeURIComponent(String(orderId)))
                .replace('{' + 'refund_id' + '}', encodeURIComponent(String(refundId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling getRefund.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling getRefund.');
            }
            // verify required parameter 'orderId' is not null or undefined
            if (orderId === null || orderId === undefined) {
                throw new Error('Required parameter orderId was null or undefined when calling getRefund.');
            }
            // verify required parameter 'refundId' is not null or undefined
            if (refundId === null || refundId === undefined) {
                throw new Error('Required parameter refundId was null or undefined when calling getRefund.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            localVarHeaderParams['x-idempotency-replayed'] = models_1.ObjectSerializer.serialize(xIdempotencyReplayed, "boolean");
            localVarHeaderParams['x-idempotency-key'] = models_1.ObjectSerializer.serialize(xIdempotencyKey, "string");
            localVarHeaderParams['x-request-id'] = models_1.ObjectSerializer.serialize(xRequestId, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'GET', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "CFRefund");
                                resolve({ cfHeaders: response.headers, cfRefund: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    /**
     * Use this API to fetch all refunds processed against an order.
     * @summary Get All Refunds for an Order
     * @param xClientId
     * @param xClientSecret
     * @param orderId
     * @param xApiVersion
     */
    getallrefundsfororder(xClientId, xClientSecret, orderId, xApiVersion, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/orders/{order_id}/refunds'
                .replace('{' + 'order_id' + '}', encodeURIComponent(String(orderId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling getallrefundsfororder.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling getallrefundsfororder.');
            }
            // verify required parameter 'orderId' is not null or undefined
            if (orderId === null || orderId === undefined) {
                throw new Error('Required parameter orderId was null or undefined when calling getallrefundsfororder.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'GET', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "Array<CFRefund>");
                                resolve({ cfHeaders: response.headers, cfRefunds: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
}
exports.RefundsApi = RefundsApi;
