// cashfree-pg-sdk-nodejs/dist/src/api/tokenVaultApi.js
"use strict";
/**
 * New Payment Gateway APIs
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2022-01-01
 * Contact: nextgenapi@cashfree.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenVaultApi = exports.TokenVaultApiApiKeys = void 0;
const request_1 = __importDefault(require("request"));
const models_1 = require("../model/models");
const apis_1 = require("./apis");
let defaultBasePath = 'https://sandbox.cashfree.com/pg';
// ===============================================
// This file is autogenerated - Please do not edit
// ===============================================
var TokenVaultApiApiKeys;
(function (TokenVaultApiApiKeys) {
})(TokenVaultApiApiKeys = exports.TokenVaultApiApiKeys || (exports.TokenVaultApiApiKeys = {}));
class TokenVaultApi {
    constructor(basePathOrUsername, password, basePath) {
        this._basePath = defaultBasePath;
        this._defaultHeaders = {};
        this._useQuerystring = false;
        this.authentications = {
            'default': new models_1.VoidAuth(),
        };
        this.interceptors = [];
        if (password) {
            if (basePath) {
                this.basePath = basePath;
            }
        }
        else {
            if (basePathOrUsername) {
                this.basePath = basePathOrUsername;
            }
        }
    }
    set useQuerystring(value) {
        this._useQuerystring = value;
    }
    set basePath(basePath) {
        this._basePath = basePath;
    }
    set defaultHeaders(defaultHeaders) {
        this._defaultHeaders = defaultHeaders;
    }
    get defaultHeaders() {
        return this._defaultHeaders;
    }
    get basePath() {
        return this._basePath;
    }
    setDefaultAuthentication(auth) {
        this.authentications.default = auth;
    }
    setApiKey(key, value) {
        this.authentications[TokenVaultApiApiKeys[key]].apiKey = value;
    }
    addInterceptor(interceptor) {
        this.interceptors.push(interceptor);
    }
    /**
     * To delete a saved instrument for a customer id and instrument id
     * @summary Delete Saved Instrument
     * @param xClientId
     * @param xClientSecret
     * @param customerId
     * @param instrumentId
     * @param xApiVersion
     */
    deleteSpecificSavedInstrument(xClientId, xClientSecret, customerId, instrumentId, xApiVersion, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/customers/{customer_id}/instruments/{instrument_id}'
                .replace('{' + 'customer_id' + '}', encodeURIComponent(String(customerId)))
                .replace('{' + 'instrument_id' + '}', encodeURIComponent(String(instrumentId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling deleteSpecificSavedInstrument.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling deleteSpecificSavedInstrument.');
            }
            // verify required parameter 'customerId' is not null or undefined
            if (customerId === null || customerId === undefined) {
                throw new Error('Required parameter customerId was null or undefined when calling deleteSpecificSavedInstrument.');
            }
            // verify required parameter 'instrumentId' is not null or undefined
            if (instrumentId === null || instrumentId === undefined) {
                throw new Error('Required parameter instrumentId was null or undefined when calling deleteSpecificSavedInstrument.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'DELETE', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "CFFetchAllSavedInstruments");
                                resolve({ cfHeaders: response.headers, cfInstrument: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    /**
     * To get all saved instruments for a customer id
     * @summary Fetch All Saved Instruments
     * @param xClientId
     * @param xClientSecret
     * @param customerId
     * @param instrumentType
     * @param xApiVersion
     * @param xIdempotencyReplayed
     * @param xIdempotencyKey
     * @param xRequestId
     */
    fetchAllSavedInstruments(xClientId, xClientSecret, customerId, instrumentType, xApiVersion, xIdempotencyReplayed, xIdempotencyKey, xRequestId, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/customers/{customer_id}/instruments'
                .replace('{' + 'customer_id' + '}', encodeURIComponent(String(customerId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling fetchAllSavedInstruments.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling fetchAllSavedInstruments.');
            }
            // verify required parameter 'customerId' is not null or undefined
            if (customerId === null || customerId === undefined) {
                throw new Error('Required parameter customerId was null or undefined when calling fetchAllSavedInstruments.');
            }
            // verify required parameter 'instrumentType' is not null or undefined
            if (instrumentType === null || instrumentType === undefined) {
                throw new Error('Required parameter instrumentType was null or undefined when calling fetchAllSavedInstruments.');
            }
            if (instrumentType !== undefined) {
                localVarQueryParameters['instrument_type'] = models_1.ObjectSerializer.serialize(instrumentType, "'card'");
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            localVarHeaderParams['x-idempotency-replayed'] = models_1.ObjectSerializer.serialize(xIdempotencyReplayed, "boolean");
            localVarHeaderParams['x-idempotency-key'] = models_1.ObjectSerializer.serialize(xIdempotencyKey, "string");
            localVarHeaderParams['x-request-id'] = models_1.ObjectSerializer.serialize(xRequestId, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'GET', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "Array<CFFetchAllSavedInstruments>");
                                resolve({ cfHeaders: response.headers, cfInstruments: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    /**
     * To get the card network token, token expiry and cryptogram for a saved instrument using instrument id
     * @summary Fetch cryptogram for saved instrument
     * @param xClientId
     * @param xClientSecret
     * @param customerId
     * @param instrumentId
     * @param xApiVersion
     * @param xIdempotencyReplayed
     * @param xIdempotencyKey
     * @param xRequestId
     */
    fetchCryptogram(xClientId, xClientSecret, customerId, instrumentId, xApiVersion, xIdempotencyReplayed, xIdempotencyKey, xRequestId, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/customers/{customer_id}/instruments/{instrument_id}/cryptogram'
                .replace('{' + 'customer_id' + '}', encodeURIComponent(String(customerId)))
                .replace('{' + 'instrument_id' + '}', encodeURIComponent(String(instrumentId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling fetchCryptogram.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling fetchCryptogram.');
            }
            // verify required parameter 'customerId' is not null or undefined
            if (customerId === null || customerId === undefined) {
                throw new Error('Required parameter customerId was null or undefined when calling fetchCryptogram.');
            }
            // verify required parameter 'instrumentId' is not null or undefined
            if (instrumentId === null || instrumentId === undefined) {
                throw new Error('Required parameter instrumentId was null or undefined when calling fetchCryptogram.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            localVarHeaderParams['x-idempotency-replayed'] = models_1.ObjectSerializer.serialize(xIdempotencyReplayed, "boolean");
            localVarHeaderParams['x-idempotency-key'] = models_1.ObjectSerializer.serialize(xIdempotencyKey, "string");
            localVarHeaderParams['x-request-id'] = models_1.ObjectSerializer.serialize(xRequestId, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'GET', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "CFCryptogram");
                                resolve({ cfHeaders: response.headers, cfCryptogram: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    /**
     * To get specific saved instrument for a customer id and instrument id
     * @summary Fetch Single Saved Instrument
     * @param xClientId
     * @param xClientSecret
     * @param customerId
     * @param instrumentId
     * @param xApiVersion
     * @param xIdempotencyReplayed
     * @param xIdempotencyKey
     * @param xRequestId
     */
    fetchSpecificSavedInstrument(xClientId, xClientSecret, customerId, instrumentId, xApiVersion, xIdempotencyReplayed, xIdempotencyKey, xRequestId, requestTimeout, webProxy, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/customers/{customer_id}/instruments/{instrument_id}'
                .replace('{' + 'customer_id' + '}', encodeURIComponent(String(customerId)))
                .replace('{' + 'instrument_id' + '}', encodeURIComponent(String(instrumentId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            // give precedence to 'application/json'
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            // verify required parameter 'xClientId' is not null or undefined
            if (xClientId === null || xClientId === undefined) {
                throw new Error('Required parameter xClientId was null or undefined when calling fetchSpecificSavedInstrument.');
            }
            // verify required parameter 'xClientSecret' is not null or undefined
            if (xClientSecret === null || xClientSecret === undefined) {
                throw new Error('Required parameter xClientSecret was null or undefined when calling fetchSpecificSavedInstrument.');
            }
            // verify required parameter 'customerId' is not null or undefined
            if (customerId === null || customerId === undefined) {
                throw new Error('Required parameter customerId was null or undefined when calling fetchSpecificSavedInstrument.');
            }
            // verify required parameter 'instrumentId' is not null or undefined
            if (instrumentId === null || instrumentId === undefined) {
                throw new Error('Required parameter instrumentId was null or undefined when calling fetchSpecificSavedInstrument.');
            }
            localVarHeaderParams['x-client-id'] = models_1.ObjectSerializer.serialize(xClientId, "string");
            localVarHeaderParams['x-client-secret'] = models_1.ObjectSerializer.serialize(xClientSecret, "string");
            localVarHeaderParams['x-api-version'] = models_1.ObjectSerializer.serialize(xApiVersion, "string");
            localVarHeaderParams['x-idempotency-replayed'] = models_1.ObjectSerializer.serialize(xIdempotencyReplayed, "boolean");
            localVarHeaderParams['x-idempotency-key'] = models_1.ObjectSerializer.serialize(xIdempotencyKey, "string");
            localVarHeaderParams['x-request-id'] = models_1.ObjectSerializer.serialize(xRequestId, "string");
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = Object.assign({ method: 'GET', qs: localVarQueryParameters, headers: localVarHeaderParams, uri: localVarPath, useQuerystring: this._useQuerystring, json: true, timeout: requestTimeout }, (webProxy && {
                proxy: webProxy
            }));
            let authenticationPromise = Promise.resolve();
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "CFFetchAllSavedInstruments");
                                resolve({ cfHeaders: response.headers, cfInstrument: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
}
exports.TokenVaultApi = TokenVaultApi;
